bn_mp_clear.c

```diff
@@ -7,7 +7,7 @@
 void mp_clear(mp_int *a)
 {
    /* only do anything if a hasn't been freed previously */
-   if (a->dp != NULL) {
+   if ((a != NULL) && (a->dp != NULL)) {
```

Comment:
a==NULL should not be handled here since it is a programming error.

Reply:
in **libtomcrypt**, we could write :
```
LTC_ARGCHK(a != NULL);
```

but sadly **libtommath** have no equivalent.

Reply:
Yes, we don't have such assertions. However the change you made masks a programming error and is therefore not good. You should revert this change.

Reply:
If you want to harden the library, what I would recommend - define MP_ASSERT(x) in tommath.h which can be optionally activated in hardened/debugging mode. Then add MP_ASSERTS where it makes sense.

Reply:
I just checked LTC, and they are doing exactly what I suggested - aborting in the case of assertion failure, but not masking the error and just returning without any message.